CondCore/SiStripPlugins/plugins/SiStripApvGain_PayloadInspector.cc

```diff
@@ -0,0 +1,420 @@
+#include "CondCore/Utilities/interface/PayloadInspectorModule.h"
+#include "CondCore/Utilities/interface/PayloadInspector.h"
+#include "CondCore/CondDB/interface/Time.h"
+
+// the data format of the condition to be inspected
+#include "CondFormats/SiStripObjects/interface/SiStripApvGain.h"
+
+#include <memory>
+#include <sstream>
+
+namespace {
+
+  /************************************************
+    1d histogram of SiStripApvGains of 1 IOV 
+  *************************************************/
+
+  // inherit from one of the predefined plot class: Histogram1D
+  class SiStripApvGainsValue : public cond::payloadInspector::Histogram1D<SiStripApvGain> {
+    
+  public:
+    SiStripApvGainsValue() : cond::payloadInspector::Histogram1D<SiStripApvGain>("SiStripApv Gains values",
+										 "SiStripApv Gains values", 200,0.2,1.8){
+      Base::setSingleIov( true );
+    }
+    
+    bool fill( const std::vector<std::tuple<cond::Time_t,cond::Hash> >& iovs ){
+      for ( auto const & iov: iovs) {
+	std::shared_ptr<SiStripApvGain> payload = Base::fetchPayload( std::get<1>(iov) );
+	if( payload.get() ){
+	 
+	  std::vector<uint32_t> detid;
+	  payload->getDetIds(detid);
+	  
+	  for (size_t id=0;id<detid.size();id++){
+	    SiStripApvGain::Range range=payload->getRange(detid[id]);
+	    for(int it=0;it<range.second-range.first;it++){
+
+	      // to be used to fill the histogram
+	      fillWithValue(payload->getApvGain(it,range));
+	      
+	    }// loop over APVs
+	  } // loop over detIds
+	}// payload
+      }// iovs
+      return true;
+    }// fill
+  };
+
+  /************************************************
+    1d histogram of means of SiStripApvGains
+    for Tracker Barrel of 1 IOV 
+  *************************************************/
+
+  // inherit from one of the predefined plot class: Histogram1D
+  class SiStripApvBarrelGainsByLayer : public cond::payloadInspector::Histogram1D<SiStripApvGain> {
+    
+  public:
+    SiStripApvBarrelGainsByLayer() : cond::payloadInspector::Histogram1D<SiStripApvGain>("SiStripApv Gains averages by Barrel layer",
+											 "Barrel layer (0-3: TIB), (4-9: TOB)",10,0,10,"average SiStripApv Gain"){
+      Base::setSingleIov( true );
+    }
+    
+    bool fill( const std::vector<std::tuple<cond::Time_t,cond::Hash> >& iovs ){
+      for ( auto const & iov: iovs) {
+	std::shared_ptr<SiStripApvGain> payload = Base::fetchPayload( std::get<1>(iov) );
+	if( payload.get() ){
+	 
+	  std::vector<uint32_t> detid;
+	  payload->getDetIds(detid);
+	  
+	  std::map<int,std::pair<float,float> > sumOfGainsByLayer;
+
+	  for (size_t id=0;id<detid.size();id++){
+
+	    int subid = int((detid[id]>>25) & 0x7);	   
```

Comment:
yuck - but I guess you can not easily get to a ES object from here? please at least use the DetId::subdetId function for subid.